lib/rules/require-default-props.js

```diff
@@ -369,9 +310,8 @@ module.exports = {
     return {
       MemberExpression: function(node) {
         const isPropType = propsUtil.isPropTypesDeclaration(node);
-        const isDefaultProp = propsUtil.isDefaultPropsDeclaration(node);
 
-        if (!isPropType && !isDefaultProp) {
+        if (!isPropType) {
```

Comment:
i'm a bit confused why the previous propTypes refactor didn't remove `isPropType`, but this defaultProps refactor is able to remove `isDefaultProp`. can you elaborate?

Reply:
Sure.

The first extraction was based on two other rules that had the most comprehensive detection functionality. However, `isRequired` detection, which is necessary for the two rules in this current PR, was not part of it. I could have added it there, but that would have blown up an already big PR.

But it is coming in the follow-up PR I mentioned above. I just didn’t want to combine it with this one, again, to keep them focused and manageable in size.

See https://github.com/alexzherdev/eslint-plugin-react/commit/438c06e3f0d71d32123583121d8c3cd8bce7cd62 and https://github.com/alexzherdev/eslint-plugin-react/commit/2d1243247c9a3b6c3cd1346d6fc74cac85947ad4 for an exclusive peek 😄

Reply:
ah ok, gotcha. so all the isPropType stuff will be able to be removed in a followup.